### ISSUE TITLE
require-await doesn't consider `await using` expressions

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#1206]
### Lint Name

`require-await`

### Code Snippet

```ts
async function a() {
  await using b = c();
}
```

### Expected Result

No lint errors.

### Actual Result

```
Async function 'a' has no 'await' expression.
Remove 'async' keyword from the function or use 'await' expression inside.
```

### Version

deno 1.37.2 (release, x86_64-unknown-linux-gnu)
v8 11.8.172.13
typescript 5.2.2
